talented-tigers/spydir/generate_page.py

```diff
@@ -0,0 +1,200 @@
+        search = wikipedia.search(page_name, results=1)
+
+        try:
+            page = wikipedia.page(search[0], auto_suggest=False)
+            print("GETTING THE FIRST SEARCH RESULT")
+
+        except IndexError:
+            # If the search yielded no results
+            print("NO RESULTS AFTER SEARCH")
+            result = "Under Construction"
+            return [result, ""]
+
+        except wikipedia.exceptions.DisambiguationError as e:
+            # If the page exists it enters is a wikipedia "disambiguation" page
+            page = wikipedia.page(e.options[0], auto_suggest=False)
+            print("THE FIRST RESULT IS A DISAMBIGUATION PAGE, USING THE FIRST LINK")
+
+    page_summary = page.summary
+    page_url = page.url
+    return [parse_result(page_summary), page_url]
+
+
+def parse_result(result):
+    wordlist = result.split()
+    information = ""
+    for word in wordlist:
+        word_token = nltk.word_tokenize(str(word))
+        if nltk.pos_tag(word_token)[0][1] == "NN" or nltk.pos_tag(word_token)[0][1] == "NNP":
+            without_punctuation = re.sub(r"[^\w\s]", "", word)
+            information += "<a href=../../page/{0}>{1}</a>".format(without_punctuation, word)
+            authorize_page(without_punctuation)
+        else:
+            information += word
+        information += " "
+
+    return information
+
+
+# TODO: Make the page update without refreshing
+def generate_gpt2(posts):
+    # Currently, the model is loaded everytime this function is called, which may be slow. Putting it outside doesnt
+    # work
+    model_name = "124M"
+    sess = gpt2.start_tf_sess()
+    gpt2.load_gpt2(sess, model_name=model_name)
+
+    for post in posts.all():
+        post.content = "Generating..."
+        post.save()
+        output = gpt2.generate(
+            sess, model_name=model_name, model_dir="models", return_as_list=True, prefix=post.title, length=100
+        )[0]
+
+        post.content = parse_result(splice_sentence(output))
+        post.save()
+
+
+def generate_business_statements(page_name, page_object):
+    # Currently, the model is loaded everytime this function is called, which may be slow. Putting it outside doesnt
+    # work
+    model_name = "124M"
+    sess = gpt2.start_tf_sess()
+    gpt2.load_gpt2(sess, model_name=model_name)
+
+    about = gpt2.generate(
+        sess, model_name=model_name, model_dir="models", return_as_list=True, prefix=f"{page_name} Co. is", length=100
+    )[0]
+    page_object.business_about = parse_result(splice_sentence(about))
+    page_object.save()
+
+    mission = gpt2.generate(
+        sess, model_name=model_name, model_dir="models", return_as_list=True, prefix="Our Mission", length=100
+    )[0]
+    page_object.business_mission = parse_result(splice_sentence(mission))
+    page_object.save()
+
+
+def splice_sentence(word):
+    cutoff = max([word.rfind("."), word.rfind("?"), word.rfind("!")])
+    return word[: cutoff + 1]
```

Comment:
overall. nice short functions. maybe a few with a lot of business logic in the generating functions. nice to see some docstrings. well written and clear